lexers/rest.lua

```diff
@@ -32,23 +32,24 @@ local underline = lpeg.Cmt(starts_line(adornment), function(_, index, adm, c)
   return pos and index - #adm + pos - 1 or nil
 end)
 -- Token needs to be a predefined one in order for folder to work.
-local title = token(l.CONSTANT, overline + underline)
+lex:add_rule('title', token(lexer.CONSTANT, overline + underline))
```

Comment:
Be careful here because 'title' is not the first rule. (See `_rules` in the legacy lexer, which has 'literal_block' first.) Either call `lex:add_rule()` all at once later, or rearrange all of the patterns and rules to match the legacy lexer rule order.